source/extensions/common/aws/signer_base_impl.cc

```diff
@@ -54,33 +67,41 @@ void SignerImpl::sign(Http::RequestHeaderMap& headers, const std::string& conten
   if (path_header == nullptr) {
     throw EnvoyException("Message is missing :path header");
   }
+
   if (credentials.sessionToken()) {
     headers.addCopy(SignatureHeaders::get().SecurityToken, credentials.sessionToken().value());
   }
   const auto long_date = long_date_formatter_.now(time_source_);
   const auto short_date = short_date_formatter_.now(time_source_);
   headers.addCopy(SignatureHeaders::get().Date, long_date);
+
+  addRegionHeader(headers, override_region);
+
   // Phase 1: Create a canonical request
   const auto canonical_headers = Utility::canonicalizeHeaders(headers, excluded_header_matchers_);
   const auto canonical_request = Utility::createCanonicalRequest(
       service_name_, method_header->value().getStringView(), path_header->value().getStringView(),
       canonical_headers, content_hash);
   ENVOY_LOG(debug, "Canonical request:\n{}", canonical_request);
+
   // Phase 2: Create a string to sign
   const auto credential_scope = createCredentialScope(short_date, override_region);
   const auto string_to_sign = createStringToSign(canonical_request, long_date, credential_scope);
   ENVOY_LOG(debug, "String to sign:\n{}", string_to_sign);
+
   // Phase 3: Create a signature
-  const auto signature = createSignature(credentials.secretAccessKey().value(), short_date,
-                                         string_to_sign, override_region);
+  const auto signature =
+      createSignature(credentials.accessKeyId().value(), credentials.secretAccessKey().value(),
+                      short_date, string_to_sign, override_region);
   // Phase 4: Sign request
   const auto authorization_header = createAuthorizationHeader(
       credentials.accessKeyId().value(), credential_scope, canonical_headers, signature);
   ENVOY_LOG(debug, "Signing request with: {}", authorization_header);
   headers.addCopy(Http::CustomHeaders::get().Authorization, authorization_header);
 }
 
-std::string SignerImpl::createContentHash(Http::RequestMessage& message, bool sign_body) const {
+std::string SignerBaseImpl::createContentHash(Http::RequestMessage& message,
+                                              const bool sign_body) const {
```

Comment:
Seems this is meaningless change?

Reply:
this was renamed to support creation of base (this method) and child classes (sigv4 and sigv4a). createContentHash is defined in base only.

Reply:
I mean the signature change. bool -> const bool.

Reply:
oh i see. let me revert that!